edgelet/edgelet-http/src/client.rs

```diff
@@ -20,67 +18,96 @@ pub trait TokenSource {
     fn get(&self, expiry: &DateTime<Utc>) -> Result<String, Self::Error>;
 }
 
-pub struct Client<S, T>
+pub trait ClientImpl: Send + Sync {
```

Comment:
See the PR description.

Reply:
yes, I should have read it more closely. :)

Reply:
The problem is the `&self` -> `&mut self`. A bunch of places relied on using `Arc<TypeThatImplsService>` and I didn't want to have to make them `Clone` / wrap `NewService` instead of `Service` (the latter especially since `NewService::new_service` is async).